internal/api/oauthserver/handlers.go

```diff
@@ -0,0 +1,184 @@
+func (s *Server) AdminOAuthServerClientRegister(w http.ResponseWriter, r *http.Request) error {
+	ctx := r.Context()
+
+	var params OAuthServerClientRegisterParams
+	if err := json.NewDecoder(r.Body).Decode(&params); err != nil {
+		return apierrors.NewBadRequestError(apierrors.ErrorCodeBadJSON, "Invalid JSON body")
+	}
+
+	// Force registration type to manual for admin endpoint
+	params.RegistrationType = "manual"
+
+	client, plaintextSecret, err := s.registerOAuthServerClient(ctx, &params)
+	if err != nil {
+		return apierrors.NewBadRequestError(apierrors.ErrorCodeValidationFailed, err.Error())
+	}
+
+	response := oauthServerClientToResponse(client, true)
+	response.ClientSecret = plaintextSecret
+
+	return shared.SendJSON(w, http.StatusCreated, response)
+}
+
+// OAuthServerClientDynamicRegister handles POST /oauth/register (OAuth 2.1 Dynamic Client Registration)
+func (s *Server) OAuthServerClientDynamicRegister(w http.ResponseWriter, r *http.Request) error {
+	ctx := r.Context()
+
+	// Check if dynamic registration is enabled
+	if !s.config.OAuthServer.AllowDynamicRegistration {
+		return apierrors.NewForbiddenError(apierrors.ErrorCodeOAuthDynamicClientRegistrationDisabled, "Dynamic client registration is not enabled")
+	}
+
+	var params OAuthServerClientRegisterParams
+	if err := json.NewDecoder(r.Body).Decode(&params); err != nil {
+		return apierrors.NewBadRequestError(apierrors.ErrorCodeBadJSON, "Invalid JSON body")
+	}
+
+	params.RegistrationType = "dynamic"
+
+	client, plaintextSecret, err := s.registerOAuthServerClient(ctx, &params)
+	if err != nil {
+		return apierrors.NewBadRequestError(apierrors.ErrorCodeValidationFailed, err.Error())
+	}
+
+	response := oauthServerClientToResponse(client, true)
+	response.ClientSecret = plaintextSecret
+
+	return shared.SendJSON(w, http.StatusCreated, response)
+}
+
+// OAuthServerClientGet handles GET /admin/oauth/clients/{client_id}
+func (s *Server) OAuthServerClientGet(w http.ResponseWriter, r *http.Request) error {
+	ctx := r.Context()
+	client := GetOAuthServerClient(ctx)
+
+	response := oauthServerClientToResponse(client, false)
+	return shared.SendJSON(w, http.StatusOK, response)
+}
+
+// OAuthServerClientDelete handles DELETE /admin/oauth/clients/{client_id}
+func (s *Server) OAuthServerClientDelete(w http.ResponseWriter, r *http.Request) error {
+	ctx := r.Context()
+	client := GetOAuthServerClient(ctx)
+
+	if err := s.deleteOAuthServerClient(ctx, client.ClientID); err != nil {
+		return apierrors.NewInternalServerError("Error deleting OAuth client").WithInternalError(err)
+	}
+
+	w.WriteHeader(http.StatusNoContent)
+	return nil
+}
+
+// OAuthServerClientList handles GET /admin/oauth/clients
+func (s *Server) OAuthServerClientList(w http.ResponseWriter, r *http.Request) error {
+	ctx := r.Context()
+	db := s.db.WithContext(ctx)
+
+	var clients []models.OAuthServerClient
+	if err := db.Q().Where("deleted_at IS NULL").Order("created_at desc").All(&clients); err != nil {
```

Comment:
```suggestion
	if err := db.Q().Where("deleted_at is null").Order("created_at desc").All(&clients); err != nil {
```

Generally Supabase convention is to use lowercase SQL, don't mind this.